### DISCUSSION TITLE
Cors issue

### DISCUSSION THREAD

**Discussion by mackelito** [simondotm/nx-firebase#188] [Q&A]
When doing a get to my function I get the classic cors message

````
Access to fetch at 'https://w3bicizk3a-uc.a.run.app/' from origin 'https://mysite.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
````

In my function
 I have added cors like so:
```
onRequest(
  // { cors: ['mysite.com'] },
  (req, res) => {
```
But that is not working so what am I missing 😁

**Comment by mackelito** [comment]
So it was all working fine locally but to make it work when deployed I had to add
```
    res.set('Access-Control-Allow-Origin', '*');
    res.set('Access-Control-Allow-Headers', 'Content-Type');
    res.set('Content-Type', 'application/json');
```

The problem now is that req.body is a empty object... again.. working fine locally but not in production :/

**Comment by simondotm** [comment]
@mackelito You can also set rewrite rules for http functions so that they are accessed via the same host as your production apps, which also solves any cors issues

More info here: https://firebase.google.com/docs/hosting/functions
